File: src/main/java/com/softwareloop/glo/Log.java
package com.softwareloop.glo;

import lombok.Getter;
import lombok.Setter;

public class Log {

    @Getter
    @Setter
    private static boolean verbose;

    //--------------------------------------------------------------------------
    // Public methods
    //--------------------------------------------------------------------------

    public static void info(String fmt, Object... args) {
        log(fmt, args);
    }

    public static void debug(String fmt, Object... args) {
        if (verbose) {
            log(fmt, args);
        }
    }

    //--------------------------------------------------------------------------
    // Private methods
    //--------------------------------------------------------------------------

    private static void log(String fmt, Object[] args) {
        System.out.printf(fmt + "%n", args);
    }

}

File: src/main/java/com/softwareloop/glo/Log.kt
package com.softwareloop.glo

object Log {

    var verbose = false

    //--------------------------------------------------------------------------
    // Public methods
    //--------------------------------------------------------------------------
    fun info(fmt: String, vararg args: Any?) {
        log(fmt, *args)
    }

    fun debug(fmt: String, vararg args: Any?) {
        if (verbose) {
            log(fmt, *args)
        }
    }

    //--------------------------------------------------------------------------
    // Private methods
    //--------------------------------------------------------------------------

    private fun log(fmt: String, vararg args: Any?) {
        System.out.println(fmt.format(*args))
    }

}